src/librustc_middle/lint.rs

```diff
@@ -230,7 +230,7 @@ pub fn struct_lint_level<'s, 'd>(
             err.allow_suggestions(false);
 
             // If this is a future incompatible lint it'll become a hard error, so
-            // we have to emit *something*. Also allow lints to whitelist themselves
+            // we have to emit *something*. Also allow lints to allowlist themselves
```

Comment:
I find the use of "allowlist" here also a bit confusing, I think this would be more precisely stated like this?

> Also, if this lint occurs in the expansion of a macro from an external crate, allow individual lints to opt-out from being reported.